scripts/import/import_variant_synonyms

```diff
@@ -147,20 +197,337 @@ sub download_PharmGKB{
+  $sth->execute();
+  my $source_id;
+  $sth->bind_columns(\$source_id);
+  $sth->fetch();
+  
+  if (!defined($source_id)) {
+    $stmt = qq{
+      INSERT INTO
+        source (
+          name,
+          description,
+          url,
+          somatic_status,
+          version
+        )
+      VALUES (
+        '$source_name',
+        '$source_description',
+        '$source_url',
+        '$source_status',
+        '$source_version'
+      )
+    };
+    $db_adaptor->dbc->do($stmt);
+    $source_id = $db_adaptor->dbc->db_handle->{'mysql_insertid'}; 
+  }
+  else {
+    $stmt = qq{
+      UPDATE
+        source 
+      SET name=?,
+        description=?,
+        url=?,
+        version=?
+      WHERE
+        source_id=?
+    };
+    my $update_source_sth =$db_adaptor->dbc->prepare($stmt);
+    $update_source_sth->execute($source_name,$source_description,$source_url,$source_version,$source_id);
+  }
+
+  return $source_id;
+}
+ 
+sub add_set_uniprot {  
+  my $db_adaptor      = shift;
+  my $set_name        = shift; 
+  my $set_description = shift;
+
+  # check if UniProt already exists 
+  my $stmt = qq{
+    SELECT
+      variation_set_id
+    FROM
+      variation_set
+    WHERE
+      name = '$set_name'
+    LIMIT 1
+  };
+
+  my $sth = $db_adaptor->dbc->prepare($stmt);
+  $sth->execute();
+  my $variation_set_id;
+  $sth->bind_columns(\$variation_set_id);
+  $sth->fetch();
+
+  if (!defined($variation_set_id)) {  
+    my $stmt_attrib = qq{
+      SELECT
+        attrib_id
+      FROM
+        attrib
+      WHERE
+        value = '$set_name'
+      LIMIT 1
+    };
+    my $sth_attrib = $db_adaptor->dbc->prepare($stmt_attrib);
+    $sth_attrib->execute();
+    my $attrib_id;
+    $sth_attrib->bind_columns(\$attrib_id);
+    $sth_attrib->fetch();
+
+    my $ins_stmt = qq{
+      INSERT IGNORE INTO
+        variation_set (
+        name,
+        description,
+        short_name_attrib_id
+        )
+      VALUES (
+        '$set_name',
+        '$set_description',
+        '$attrib_id' 
+      )
+    };
+
+    $db_adaptor->dbc->do($ins_stmt); 
+    $variation_set_id = $db_adaptor->dbc->db_handle->{'mysql_insertid'};
+ 
+  } 
+  
+  return $variation_set_id; 
+}
+
+sub add_synonyms_uniprot {
+  my $synonyms         = shift;
+  my $variation_ids    = shift;
+  my $source_id        = shift;
+  my $variation_set_id = shift; 
+  my $db_adaptor       = shift;
+
+  # If we actually didn't get any synonyms, just return
+  return if (!defined($synonyms) || !scalar(keys(%{$synonyms})));
+  
+  # Some prepeared statements needed for inserting the synonyms into database
+  my $ins_stmt = qq{
+    INSERT IGNORE INTO
+      variation_synonym (
+      variation_id,
+      source_id,
+      name
+      )
+    VALUES (
+      ?,
+      $source_id,
+      ?
+    )
+  };
+
+  my $stmt_update = qq{
+    UPDATE
+      variation_feature 
+    SET 
+      variation_set_id = CONCAT(variation_set_id,',$variation_set_id')  
```

Comment:
CONCAT will return NULL if variation_set_id IS NULL, currently it not allowed to be NULL but this may change in future see ENSVAR-1091

Reply:
I'm going to write a note in ENSVAR-1091 about this script and leave it for now like this.

Reply:
Can the update to variation_set_id be done with either a
CONCAT(ifnull(variation_set_id,''), ',$variation_set_id')
or
CONCAT_WS(',', variation_set_id, '$variation_set_id')
instead of a check and then an insert which may have race conditions

Reply:
Thanks for the suggestion!